unstructured/ingest/connector/confluence.py

```diff
@@ -0,0 +1,164 @@
+    for more info on the api_token.
+    """
+
+    user_email: str
+    api_token: str
+    url: str
+
+
+@dataclass
+class ConfluenceFileMeta:
+    """Metadata specifying:
+    id for the confluence space that the document locates in,
+    and the id of document that is being reached to.
+    """
+
+    space_id: str
+    document_id: str
+
+
+@dataclass
+class ConfluenceIngestDoc(IngestDocCleanupMixin, BaseIngestDoc):
+    """Class encapsulating fetching a doc and writing processed results (but not
+    doing the processing!).
+
+    Current implementation creates a Confluence connection object
+    to fetch each doc, rather than creating a it for each thread.
+    """
+
+    config: SimpleConfluenceConfig
+    file_meta: ConfluenceFileMeta
+
+    # TODO: remove one of filename or _tmp_download_file, using a wrapper
+    @property
+    def filename(self):
+        return (
+            Path(self.standard_config.download_dir)
+            / self.file_meta.space_id
+            / f"{self.file_meta.document_id}.txt"
+        ).resolve()
+
+    @property
+    def _output_filename(self):
+        """Create output file path based on output directory, space id and document id"""
+        output_file = f"{self.file_meta.document_id}.json"
+        return Path(self.standard_config.output_dir) / self.file_meta.space_id / output_file
+
+    @requires_dependencies(["atlassian"])
+    @BaseIngestDoc.skip_if_file_exists
+    def get_file(self):
+        logger.debug(f"Fetching {self} - PID: {os.getpid()}")
+
+        # TODO: instead of having a separate connection object for each doc,
+        # have a separate connection object for each process
+        confluence = Confluence(
+            self.config.url,
+            username=self.config.user_email,
+            password=self.config.api_token,
+        )
+
+        result = confluence.get_page_by_id(page_id=self.file_meta.document_id, expand="body.view")
+        self.document = result["body"]["view"]["value"]
+        self.filename.parent.mkdir(parents=True, exist_ok=True)
+        with open(self.filename, "w", encoding="utf8") as f:
+            f.write(self.document)
+
+
+@requires_dependencies(["atlassian"])
+@dataclass
+class ConfluenceConnector(ConnectorCleanupMixin, BaseConnector):
+    """Fetches the body field from all documents from all spaces in a Confluence Cloud instance."""
+
+    config: SimpleConfluenceConfig
+
+    def __init__(
+        self,
+        standard_config: StandardConnectorConfig,
+        config: SimpleConfluenceConfig,
+    ):
+        super().__init__(standard_config, config)
+
+    @requires_dependencies(["atlassian"])
+    def initialize(self):
+        self.confluence = Confluence(
+            url=self.config.url,
+            username=self.config.user_email,
+            password=self.config.api_token,
+        )
+
+    @requires_dependencies(["atlassian"])
+    def _get_all_space_ids(self, start: int = 0, limit: int = 500, expand=None):
```

Comment:
Might be good to expose the `limit` variable to the user

Reply:
great idea!